extensions/hazelcast-client/deployment/src/main/java/io/quarkus/hazelcast/client/deployment/HazelcastClientProcessor.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    private void registerCustomCredentialFactories() {
+        registerTypeHierarchy(
+                reflectiveClassHierarchies,
+                com.hazelcast.security.ICredentialsFactory.class);
+
+        reflectiveClasses.produce(
+                new ReflectiveClassBuildItem(false, false, DefaultCredentialsFactory.class));
+    }
+
+    private void registerSSLUtilities() {
+
+        registerTypeHierarchy(
+                reflectiveClassHierarchies,
+                com.hazelcast.nio.ssl.SSLContextFactory.class);
+        reflectiveClasses.produce(
+                new ReflectiveClassBuildItem(false, false, BasicSSLContextFactory.class));
+    }
+
+    private void registerUserImplementationsOfSerializableUtilities() {
+        registerTypeHierarchy(reflectiveClassHierarchies,
+                DataSerializable.class,
+                DataSerializableFactory.class,
+                PortableFactory.class,
+                Serializer.class);
+    }
+
+    private void registerICMPHelper() {
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(false, false, ICMPHelper.class));
+        resources.produce(new NativeImageResourceBuildItem(
+                "lib/linux-x86/libicmp_helper.so",
+                "lib/linux-x86_64/libicmp_helper.so"));
+        reinitializedClasses.produce(new RuntimeReinitializedClassBuildItem(ICMPHelper.class.getName()));
+    }
+
+    private void registerXMLParsingUtilities() {
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(false, false,
+                "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",
+                "com.sun.xml.bind.v2.ContextFactory",
+                "com.sun.xml.internal.stream.XMLInputFactoryImpl",
+                "com.sun.org.apache.xpath.internal.functions.FuncNot",
+                "com.sun.xml.internal.bind.v2.ContextFactory",
+                "com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl",
+                "com.sun.org.apache.xerces.internal.impl.dv.xs.SchemaDVFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.datatype.DatatypeFactoryImpl"));
+
+        bundles.produce(new NativeImageResourceBundleBuildItem(
+                "com.sun.org.apache.xml.internal.serializer.utils.SerializerMessages"));
+        bundles.produce(new NativeImageResourceBundleBuildItem(
+                "com.sun.org.apache.xerces.internal.impl.msg.XMLMessages"));
+        bundles.produce(new NativeImageResourceBundleBuildItem(
+                "com.sun.org.apache.xerces.internal.impl.msg.XMLSchemaMessages"));
+        bundles.produce(new NativeImageResourceBundleBuildItem(
+                "com.sun.org.apache.xerces.internal.impl.xpath.regex.message"));
+
+        resources.produce(new NativeImageResourceBuildItem(
+                "com/sun/org/apache/xml/internal/serializer/output_xml.properties"));
```

Comment:
It looks like they are already handled in the `jaxb` extension? Try depending your extension on it and see if that's sufficient.

Reply:
Unfortunately, it doesn't really help - I will investigate and ideally fix the issues in `jaxb` so that we can reuse it

Reply:
Fix'd: https://github.com/quarkusio/quarkus/pull/6863/files